browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -20,6 +26,66 @@
 #include "extensions/browser/extension_registry.h"
 #endif
 
+namespace {
+
+class CtrlReleaseHandler : public ui::EventHandler {
+ public:
+  explicit CtrlReleaseHandler(BraveTabStripModel* model,
+                              BraveBrowserView* browser_view)
+      : model_(model), browser_view_(browser_view) {}
+  ~CtrlReleaseHandler() override = default;
+
+ private:
+  // ui::EventHandler overrides:
+  void OnKeyEvent(ui::KeyEvent* event) override {
+    if (event->key_code() == ui::VKEY_CONTROL &&
+        event->type() == ui::ET_KEY_RELEASED) {
+      // Ctrl key was released, stop the MRU cycling
+
+      // Remove event handler
+#if defined(OS_MACOSX)
+      if (browser_view_->GetWidget()->GetRootView())
+        browser_view_->GetWidget()->GetRootView()->RemovePreTargetHandler(this);
+#else
+      if (browser_view_->GetWidget()->GetNativeWindow())
+        browser_view_->GetWidget()->GetNativeWindow()->RemovePreTargetHandler(
+            this);
+#endif
+
+      model_->StopMRUCycling();
+    } else if (!((event->key_code() == ui::VKEY_TAB &&
```

Comment:
Still curious why other keys should be blocked?

Reply:
Because I assume that nothing else can be done while cycling (closing tab with shortcut, reordering tabs, etc..)
Also, Alt-tab implementation for ChromeOS does the same: https://chromium.googlesource.com/chromium/src/+/refs/heads/master/ash/wm/window_cycle_event_filter.cc#39

Btw, I just notice that we can still close tabs with the mouse while cycling. This leads to crashes, let block all clicks on the window.

Reply:
I see. How about stop mru cycling when browser gets mouse event ?

Reply:
+1 for using firefox behaviour for reference.

Reply:
Good point!